### ISSUE TITLE
Upgrade Kubernetes-external-secret

### ISSUE THREAD

**Issue by j-zimnowoda** [redkubes/otomi-core#743]
**Is your feature request related to a problem? Please describe.** 

The [KES](https://github.com/external-secrets/kubernetes-external-secrets) project is deprecated, because it has been moved to https://github.com/external-secrets/external-secrets. We should start using the new version.

**Describe the solution you'd like** 

Replace the `kubernetes-external-secrets` helm chart with the new one: https://github.com/external-secrets/external-secrets/tree/main/deploy/charts/external-secrets
Upgrade corresponding values.

**Comment by Morriz** [comment]
@staticvoid255 @j-zimnowoda please be aware the just replacing the chart is not gonna work (I tried). The "new" chart is actually another chart, with another interface. Please investigate what is needed and make a migration plan here first.

**Comment by staticvoid255** [comment]
The reason this was picked up is that GU are encountering flaky syncing and must restart the kubernetes-external-secrets pod manually each time new secrets are created - this has proven to be highly disruptive as very few of us have access to do so, and I am receiving many comments about this every day.

I found that external-secrets exposes a generic `SecretStore` object, rather than specifying the backend in Helm values. The `ExternalSecret` definition has also changed to take a reference to this new `SecretStore`, so user secrets created with the old version will need to be migrated for sure.

Tasks for now:

1. `SecretStore` needs to be created after or during external-secrets install configuring Vault as the backend - how should I do this? I could add it to the external-secrets helm chart and pass in values through the .gotmpl - this seems ok to me as we don't install external-secrets without Vault, and vice versa, unless we plan on supporting such cases in the future. 
2. Migration process for existing secrets / secrets created with old version - need advice, have no suggestions, what I do know is that at the very least `ExternalSecret`s require a `secretStoreRef` 

There are other secrets properties like `lifecycle` that are new or have changed - I am still looking into this.

I believe it would be easier to just update to the newest release of kubernetes-external-secrets (8.5.5 vs our 6.1.0) and see if that solves the need for manual restarting, then migrate to external-secrets at a later date when there's less pressure.

**Comment by Morriz** [comment]
So just restarting the pod helped, and that is a classic case of "have you tried turning ...?". And we were already at `8.5.5` we saw, so that leaves upgrading to the new controller. I saw no special labels on the resulting native k8s secrets, so I think I can carefully assume that they are only targeted by name, and we don't have to worry about pre-adoption of any resources. So the drop-in replacement we will realize is not impactful from an integration perspective.

**Comment by staticvoid255** [comment]
We did turn it on and off again - that may have resulted in a successful sync **this time**, but does not actually solve issue and is not sustainable. The real issue is the default polling interval being set to 10 seconds resulting in overly aggressive pinging of the kube-api given the number of secrets. Recommend specifying an interval of 60+ seconds (5-10 minutes is common) in our defaults - I will create a PR for this.

**Comment by staticvoid255** [comment]
I have been looking into this further with Jordi - we have encountered more issues where after creating a kubernetes-external-secrets.yaml file and specifying a polling value (and nothing else), it breaks. This issue is because our Helm chart is 6.1.0 (should be 8.5.5), but our image version is 8.5.5, hence why I thought our image version would be 6.1.0 because of their release strategy. Regardless, in the end this is because someone carelessly updated the image version in values without updating the Helm chart.

`➜  ~ ont -n vault logs -f kubernetes-external-secrets-685b59ff9b-stngd                                                                                                                                                >> team-ritense-otomi-aks-ont

> kubernetes-external-secrets@8.5.5 start /app
> ./bin/daemon.js

{"level":30,"message_time":"2022-06-21T14:41:03.958Z","pid":17,"hostname":"kubernetes-external-secrets-685b59ff9b-stngd","payload":{},"msg":"loading kube specs"}
{"level":30,"message_time":"2022-06-21T14:41:06.443Z","pid":17,"hostname":"kubernetes-external-secrets-685b59ff9b-stngd","payload":{},"msg":"successfully loaded kube specs"}
{"level":30,"message_time":"2022-06-21T14:41:06.445Z","pid":17,"hostname":"kubernetes-external-secrets-685b59ff9b-stngd","payload":{},"msg":"verifiying CRD is installed"}
{"level":50,"message_time":"2022-06-21T14:41:16.545Z","pid":17,"hostname":"kubernetes-external-secrets-685b59ff9b-stngd","payload":{},"msg":"CRD installation check failed, statusCode: undefined"}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kubernetes-external-secrets@8.5.5 start: `./bin/daemon.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the kubernetes-external-secrets@8.5.5 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2022-06-21T14_41_16_640Z-debug.log`

**Comment by staticvoid255** [comment]
also this, also related to using old helm chart:
```
camunda                     2m44s       ERROR, secrets "camunda" is forbidden: User "system:serviceaccount:vault:kubernetes-external-secrets" cannot get resource "secrets" in API group "" in the namespace "team-utrecht"                     91d
fluxcd-tenant-key           2m42s       ERROR, secrets "fluxcd-tenant-key" is forbidden: User "system:serviceaccount:vault:kubernetes-external-secrets" cannot get resource "secrets" in API group "" in the namespace "team-utrecht"           91d
gitlab-helm-internal-repo   2m42s       ERROR, secrets "gitlab-helm-internal-repo" is forbidden: User "system:serviceaccount:vault:kubernetes-external-secrets" cannot get resource "secrets" in API group "" in the namespace "team-utrecht"   91d
oauth2-proxy                2m43s       ERROR, secrets "oauth2-proxy" is forbidden: User "system:serviceaccount:vault:kubernetes-external-secrets" cannot get resource "secrets" in API group "" in the namespace "team-utrecht"                7d23h
```

This is fixed in the newest helm chart release by specifying the "get" verb for serviceaccount access to secrets in rbac.yaml

**Comment by Morriz** [comment]
> Regardless, in the end this is because someone carelessly updated the image version in values without updating the Helm chart.

You may diff the latest 8.5.5 chart against what we have now. If there are critical changes (like that rbac setting), please apply those diffs.

**Comment by Morriz** [comment]
fixed in v0.16.9
